### ISSUE TITLE
kong proxy listen cannot use reuseport?

### ISSUE THREAD

**Issue by so2bin** [Kong/kong#4452]
### Summary

Kong start with error when I config proxy_listen with reuseport like this: 
```
KONG_PROXY_LISTEN: "0.0.0.0:8000 reuseport"
```
the error is:
```
Error: proxy_listen must be of form: [off] | <ip>:<port> [ssl] [http2] [proxy_protocol] [transparent], [... next entry ...]
```

### Steps To Reproduce

1. Configurate `kong.conf` with `proxy_listen: 0.0.0.0:8000 reuseport`

Thanks

**Comment by dndx** [comment]
Looks like we do not allow the `reuseport` flag when parsing listeners in https://github.com/Kong/kong/blob/master/kong/conf_loader.lua#L848, which is definitely worth supporting.

Personally I think the `reuseport` option should be enabled by default. It significantly improves load balancing between multiple workers when accepting incoming connections. If the O/S does not support the `SO_REUSEPORT` flag NGINX will simply [print out an error message](http://lxr.nginx.org/source/src/http/ngx_http_core_module.c#3952) during startup (which should only happen on very old Linux system).

More readings:
* https://lwn.net/Articles/542629/
* https://blog.cloudflare.com/the-sad-state-of-linux-socket-balancing/

**Comment by thibaultcha** [comment]
PRs to accept `reuseport` in the listeners parsing are most definitely welcome!

**Comment by jeremyjpj0916** [comment]
Yeah I run with reuseport and deferred on my listeners but hardcoded in a custom kong nginx conf file. Glad to see others giving it a go and would be good for kong to be able to accept theses as configurable parameters too. I am all about some nginx conf optimizations I read off random blogs and nginx docs 😆.

**Comment by bungle** [comment]
also ´so_keepalive` etc. The current `ip:port` is also more limited than what Nginx allows, such as `*:80`. Also now it looks like Nginx started to support port-ranges such as `*:80-90`. Maybe the best option is just not try to validate these inputs and just accept array of strings. Nginx will cry out loud ehen starting if it is not valid.

**Comment by dndx** [comment]
This has been released in [Kong 1.3.0rc1](https://discuss.konghq.com/t/kong-1-3-0rc1-available-for-testing/4092). Please give it a try!
